### PR TITLE
Review and Updates to BYOD Pattern

### PR DESCRIPTION
**Pull request by SamBH**
Review and updates to the BYOD pattern as discussed with @oobedan

### CODE REVIEW

File: patterns/byod.md [hunk 1]
```diff
@@ -0,0 +1,579 @@
...
+
+### Mobile device access (BYOD Option 1)
+
+The Mobile device access option will leverage:
+
+* Azure Active Directory Multi-Factor Authentication (MFA) and Conditional Access policies
+
+* Defender for Cloud Apps to prevent data leakage and spills
+
+* Intune - Application Protection Policies.
```
Comment (zedzedzed) [inline patterns/byod.md:149]: Could you please remove the extra spaces between the list items as these will convert to HTML open/close ULs

File: patterns/byod.md [hunk 2]
```diff
@@ -0,0 +1,579 @@
...
+* To protect the data in the event that a device is lost or stolen or if the employee leaves the Agency, Selective Wipe capability can be used to wipe the corporate data in managed apps from the device. Intune app protection policies can help prevent data contamination between non managed apps on a client device.
+* Agencies should provide some guidance or user guides to their users on how to install the Office applications on their devices and authenticate to avoid confusion. Note, these actions and out-of-the-box setup are generally automated within a supervised, managed deployment. 
+
+#### Azure AD Identity Protection
+
+Azure AD Identity Protection is an Azure AD feature that protects the identity through:
+
+* automation of detection and remediation of identity-based risks,
+* includes additional challenge to user based on risk detection in real time, and
+* Provides a portal for identify based investigation.
```
Comment (zedzedzed) [inline patterns/byod.md:176]: `identify` is incorrect, is it meant to be `identity`? Also lower case `provides`

File: patterns/byod.md [hunk 3]
```diff
@@ -0,0 +1,579 @@
...
+Defender for Cloud Apps | Cloud App security Session policies will restrict access to data within browser sessions | Provides a Zero Trust method to restrict access to leaking of data. 
+Intune app protection policies | App protection polices will be deployed to the MAM identity groups without enrollment required (MAM-WE) | Provides some protection against Agency data within a managed app. Protection will require a passcode within the app, control sharing of data between apps and prevent copy of data to personal locations. This allows separation between corporate and personal data to address the intent of ISM security control 1400. 
+Azure AD Identity Protection | Blueprint configuration will remain in place enabling Enable the sign-in risk policy and user risk policy within the Azure AD tenant | Provide reporting of detected suspicious sign-in activity based on defined MFA, sign-in risk and user risk policies for increased security and automated remediation steps such as requiring a password change or MFA challenge when authentication is suspicious.
```
Comment (zedzedzed) [inline patterns/byod.md:197]: `Blueprint configuration will remain in place enabling Enable the sign-in risk policy and user risk policy within the Azure AD tenant`

Maybe some missing code blocks or is it bad sentence joining?

File: patterns/byod.md [hunk 4]
```diff
@@ -0,0 +1,579 @@
...
+Azure AD Identity Protection | Blueprint configuration will remain in place enabling Enable the sign-in risk policy and user risk policy within the Azure AD tenant | Provide reporting of detected suspicious sign-in activity based on defined MFA, sign-in risk and user risk policies for increased security and automated remediation steps such as requiring a password change or MFA challenge when authentication is suspicious. 
+Multi Factor Authentication | Multi Factor authentication for this use-case will always be enforced. | As per the Blueprint baselines defined within the [platform design](https://desktop.gov.au/blueprint/platform.html). 
+
+#### BYOD Option 1 configuration guide
+
+The configuration options provided here assume that the PROTECTED utility has been implemented.
```
Comment (zedzedzed) [inline patterns/byod.md:202]: Replace `PROTECTED utility` with `Protected Utility`

File: patterns/byod.md [hunk 5]
```diff
@@ -0,0 +1,579 @@
...
+
+##### Conditional Access Polices
+
+The following updates are required to the following Conditional Access Policies as deployed standard within the Blueprint.
+
+`Azure AD Conditional Access > Policies`
+
+| Item                                            | Value                                                        |
+| ----------------------------------------------- | ------------------------------------------------------------ |
+| Existing policy name: GRANT - iOS Device access | This policy requires iOS device state to be compliant which requires enrollment. The update required to this policy is the exclude the MAM users group.<br /><br />Assignments > Exclude > rol-Agency-MAMUsers |
```
Comment (zedzedzed) [inline patterns/byod.md:232]: Replace `policy is the exclude` with `policy is to exclude`

File: patterns/byod.md [hunk 6]
```diff
@@ -0,0 +1,579 @@
...
+
+The following is a new Conditional Access Policy to allow MAM access for BYOD option 2.
```
Comment (zedzedzed) [inline patterns/byod.md:234]: Are we still talking about option 1 rather than option 2 here?

File: patterns/byod.md [hunk 7]
```diff
@@ -0,0 +1,579 @@
...
+    * Require all the selected controls: `Yes`
+    * Require one of the selected controls: `No`
+* Session
+  * Use app enforced restrictions: `True`
+  * Use Conditional Access App Control: `True > Custom policy`
+  * Sign-in frequency: `8 Hours`
+  * Persistent browser session: `False`
+* Enable policy: `On`
+
+The following is a new Conditional Access Policy to allow block unapproved apps for MAM users on iOS and Android.
```
Comment (zedzedzed) [inline patterns/byod.md:299]: `to allow block unapproved apps` I think this is meant to be `to block unapproved apps`?

File: patterns/byod.md [hunk 8]
```diff
@@ -0,0 +1,579 @@
...
+| Offline grace period                                     | 30 days  (Wipe data)                                         |
+| Jailbroken/rooted devices                                | Block access                                                 |
+| Min OS version                                           | 12.0 (Block access)                                          |
+| Disabled account                                         | Block access                                                 |
+| **Assignments**                                          |                                                              |
+| Included groups                                          | rol-Agency-MAMUsers                                          |
+
+### Virtual desktop (BYOD Option 2)
+
+The Virtual desktop (Azure Virtual Desktop) device access option will leverage:
```
Comment (zedzedzed) [inline patterns/byod.md:533]: In this instance, `The Virtual desktop` can be `The virtual desktop`

File: patterns/byod.md [hunk 9]
```diff
@@ -0,0 +1,579 @@
...
+
+* Access all M365 applications, from within the virtual desktop only
+* Access all other corporate applications, from within the virtual desktop only
+* Prevent copy, paste, saving of files and printing on the Mobile Device.
+
+The recommendation to implement this option is to utilise the [Azure Virtual Desktop Blueprint](https://desktop.gov.au/patterns/avd.html) pattern. The solution requires architectural decisions to be made that are outside the scope of this pattern. This BYOD option is more difficult to implement than the MAM BYOD option 1, but offers a less complex implementation than other virtual desktop solutions.
+
+The pattern recommends the following:
+
+* A suitably configured Azure subscription is available, which may include connectivity to the Agencies IT premises (e.g. data centre) where connectivity to hybrid Active Directory or corporate apps is required.
```
Comment (zedzedzed) [inline patterns/byod.md:548]: Replace `Agencies IT` with `Agency's IT`

File: patterns/byod.md [hunk 10]
```diff
@@ -0,0 +1,579 @@
...
+* Windows Desktop OS multi-session operation system is used to offer greater supportability and cost savings than that of a single instance virtual machines.
```
Comment (zedzedzed) [inline patterns/byod.md:549]: Replace `operation` with `operating`

File: patterns/byod.md [hunk 11]
```diff
@@ -0,0 +1,579 @@
...
+* Copy and paste actions, file copy and print are prevented within RDP client properties and group policy to prevent virtual channels being used to copy data in and out of the PROTECTED boundary (the virtual desktop).
+* Multifactor authentication is enforced with session controls to prevent multiple day use of the MFA token.
+* Identity Protection is configured to allow detection of risky sign-ins.
+
+**Mobile devices** or **personal devices** in the context of this BYOD option are consumer desktop devices using MacOS or Windows, or devices provided by another government Agency (cross-Agency working use-cases).
+
+#### Data segregation
+
+AVD allows users to access to Office 365 applications that are hosted on a Windows Virtual Desktop, including any hybrid applications provided these are configured appropriately within the Azure environment. These applications are running within a virtual desktop so the user operating the client device can only interact with the application inside the virtual machine, but cannot extract or upload data to and from the local endpoint when configured appropriately.
```
Comment (zedzedzed) [inline patterns/byod.md:559]: Replace `AVD allows users to access to Office` with `AVD allows users to access Office`

Maybe replace the following:

`These applications are running within a virtual desktop so the user operating the client device can only interact with the application inside the virtual machine, but cannot extract or upload data to and from the local endpoint when configured appropriately.`

with

`The user interacts with the applications within the virtual desktop preventing the user from extracting corporate data to the local physical device.`

File: patterns/byod.md [hunk 12]
```diff
@@ -0,0 +1,579 @@
...
+ Virtual desktop (Option2) | Provided through the Azure Virtual Desktop service, this provides a full Windows working environment similar to the corporate desktop. Access is available from a user's personal Windows or Mac device to access the virtual desktop.<br /><br />Remote working within another government Agency or partner environment.<br /><br />Legacy or non-cloud corporate applications (hybrid) can operate with this use-case.<br /><br />Use case requires always-on Internet connection. | More difficult | Greater compliance with PROTECTED-level ISM controls and the Essential Eight<br /><br />Underlying device used to access AVD, screen scraping and physical security 
+
+The following process can be followed to determine which pattern option to follow.
+
+![BYO Flow Chart](/assets/images/patterns/flow.png)
+
+### Security Considerations
+
+Introducing BYOD carries some risk for the Agency. While some controls can be addressed through a technical control, others require user training and Agency policy (for examples, Standard Operating Procedures (SOPs)) in place to ensure responsibilities are met and understood, and the appropriate mitigations are in place.
```
Comment (zedzedzed) [inline patterns/byod.md:97]: `for example` - remove trailing s

**Review by zedzedzed — COMMENTED** [review]
For the flow chart (flow.png):
* for decision point "Assess use case for iOS and Android access", I think it needs a no decision to finish
* "Is iOS and Android access..." should be "Are iOS and Android access..."

For byodoptions.png:
* I wonder if it should have column/row labels to help explain the boxes
* What do the colours mean?

Could we please try to keep blueprint with lowercase b unless starting a new sentence?

**Comment by SamBH** [comment]
Closed PR for desktop.gov.au and opened for @oobedan's fork.

**Pull request closed without merging**